src/upload/catalog/controller/extension/payment/webpay_rest.php

```diff
@@ -245,29 +264,40 @@ private function failView() {
         $data['reject_text'] = $this->session->data['reject_text'];
         $data['continue'] = $this->url->link('checkout/checkout');
 
-        $this->response->setOutput($this->load->view('extension/payment/webpay_fail', $data));
+        $this->response->setOutput($this->load->view('extension/payment/webpay_rest_fail', $data));
     }
 
     private function successView() {
 
         $result = $this->session->data['result'];
-
-        if (!is_array($result)) {
+        if ($result instanceof stdClass) {
             $result = json_decode(json_encode($result));
         }
-
+        if ($result instanceof TransactionCommitResponse) {
```

Comment:
I thought you could `json_encode` that

Reply:
I could check. The previous if (!is_array($result)) { did a `json_decode(json_encode(...))` but for some reason it failed so i changed that conditional case.